docs/DeveloperGuide.adoc

```diff
@@ -358,6 +359,138 @@ image::UndoRedoActivityDiagram.png[width="650"]
+The detailed code is shown below:
+----
+public boolean test(Person person) {
+    return keywords.stream()
+            .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(person.getEmail().value, keyword))
+            && keywords.stream()
+            .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(person.getAddress().value, keyword))
+            && keywords.stream()
+            .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(person.getPhone().value, keyword));
+}
+----
+[NOTE]
+If a `KEYWORD` that is not a name, or a name that is not registered in TuitionCor is entered, TuitionCor will return an empty list.
+
+==== Future Improvements
+
+* Currently, match can only be done one client at a time. future implementations could be updating it to match all the possible combinations.
+* Currently, when matching a name or a word that is not stored in TuitionCor, an empty list is returned. This could be misleading since the user would not know its a typo or there is simply no match. Future implementation could help to resolve such issue.
+* Currently, a match only takes place when all the data of another client fits the data of the client entered. Future implementation could return a list of clients that ranks from highest match to lowest match. This provide more flexible choices to our clients.
+// end::match[]
+
+// tag::sorting[]
+=== Sort feature
+==== Current Implementation
+
+The sort command is facilitated by a SortCommandParser which implements Parser<SortCommand>. The sort function supports sorting of different fields, namely location, grade, subject and name, which updates the list according to the sorting method.
+
+Different types of sorting methods extends abstract class SortCommand which extends Command resides in Logic as shown in the model below.
+
+image::SortCommandClassDiagram.PNG[width="800"]
+
+The different type of sort command are implemented this way:
+[source,java]
+----
+public class SortByNameCommand extends SortCommand {
+     @Override
+     public CommandResult execute() {
+         // ... list logic ...
+     }
+}
+
+public class SortByLocationCommand extends SortCommand {
+    @Override
+    public CommandResult execute() {
+        // ... list logic ...
+    }
+}
+
+public class SortByGradeCommand extends SortCommand {
+    @Override
+    public CommandResult execute() {
+        // ... list logic ...
+    }
+}
+
+public class SortBySubjectCommand extends SortCommand {
+    @Override
+    public CommandResult execute() {
+        // ... list logic ...
+    }
+}
+----
+
+When the user wants to sort the specific list according to his/her needs, SortCommandParser will parse the user input and decide which type of SortCommand to invoke.
+An example of the input required "sort tl" which sorts tutor's list by location.
+
+[NOTE]
+If the user input format is invalid, an exception will be thrown
```

Comment:
Does the exception lead to an error message being shown to the user or??

Include code.

Reply:
Will update